src/SFA.DAS.Apim.Developer.Application/AzureApimManagement/Services/SubscriptionService.cs

```diff
@@ -146,5 +146,28 @@ public async Task<IEnumerable<Subscription>> GetUserSubscriptions(string interna
 
             return returnList;
         }
+
+        public async Task DeleteSubscription(string internalUserId,
+            ApimUserType apimUserType, string productName)
+        {
+            var subscriptionId = $"{apimUserType}-{internalUserId}-{productName}";
+            var deleteSubscriptionRequest = new DeleteSubscriptionRequest(subscriptionId);
+            var apiResponse = await _azureApimManagementService.Delete<string>(deleteSubscriptionRequest);
+            
+            await _apimSubscriptionAuditRepository.Insert(
+                new ApimSubscriptionAudit
+                {
+                    Action = "deleted subscription",
+                    ProductName = productName,
+                    UserId = internalUserId,
+                    ApimUserType = (short)apimUserType
+                });
```

Comment:
Awesome to add this in so we have some visibility of keys being deleted. Will get this added to the grafana dashboard